Reactive resteasy native exception

## Describe the bug
When running reactive resteasy in native mode and accessing an endpoint with insufficient right a forbidden exception is thrown and is catched by `io.quarkus.resteasy.reactive.server.runtime.exceptionmappers.ForbiddenExceptionMapper`that throws an exception

### Expected behavior
This should work flowlessly 

### Actual behavior
An exception is thrown with following : 

`
Caused by: java.lang.InstantiationException: Type io.quarkus.resteasy.reactive.server.runtime.exceptionmappers.ForbiddenExceptionMapper` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.
`

### Quarkus version : 1.13.0.Final

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveCDIProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveCDIProcessor.java
@@ -21,6 +21,7 @@ import io.quarkus.arc.processor.BuiltinScope;
 import io.quarkus.arc.processor.DotNames;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.resteasy.reactive.common.deployment.ResourceScanningResultBuildItem;
 import io.quarkus.resteasy.reactive.server.runtime.QuarkusContextProducers;
 import io.quarkus.resteasy.reactive.spi.DynamicFeatureBuildItem;
@@ -84,6 +85,7 @@ public class ResteasyReactiveCDIProcessor {
 
     @BuildStep
     void additionalBeans(List<DynamicFeatureBuildItem> additionalDynamicFeatures,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer,
             List<JaxrsFeatureBuildItem> featureBuildItems,
             BuildProducer<AdditionalBeanBuildItem> additionalBean) {
 
@@ -91,11 +93,17 @@ public class ResteasyReactiveCDIProcessor {
         for (DynamicFeatureBuildItem dynamicFeature : additionalDynamicFeatures) {
             if (dynamicFeature.isRegisterAsBean()) {
                 additionalProviders.addBeanClass(dynamicFeature.getClassName());
+            } else {
+                reflectiveClassBuildItemBuildProducer
+                        .produce(new ReflectiveClassBuildItem(true, false, false, dynamicFeature.getClassName()));
             }
         }
-        for (JaxrsFeatureBuildItem dynamicFeature : featureBuildItems) {
-            if (dynamicFeature.isRegisterAsBean()) {
-                additionalProviders.addBeanClass(dynamicFeature.getClassName());
+        for (JaxrsFeatureBuildItem feature : featureBuildItems) {
+            if (feature.isRegisterAsBean()) {
+                additionalProviders.addBeanClass(feature.getClassName());
+            } else {
+                reflectiveClassBuildItemBuildProducer
+                        .produce(new ReflectiveClassBuildItem(true, false, false, feature.getClassName()));
             }
         }
         additionalBean.produce(additionalProviders.setUnremovable().setDefaultScope(DotNames.SINGLETON).build());

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveScanningProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveScanningProcessor.java
@@ -51,6 +51,7 @@ import io.quarkus.deployment.Capability;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.index.IndexingUtil;
 import io.quarkus.resteasy.reactive.common.deployment.ApplicationResultBuildItem;
 import io.quarkus.resteasy.reactive.common.deployment.ResourceInterceptorsContributorBuildItem;
@@ -99,6 +100,7 @@ public class ResteasyReactiveScanningProcessor {
     public ExceptionMappersBuildItem scanForExceptionMappers(CombinedIndexBuildItem combinedIndexBuildItem,
             ApplicationResultBuildItem applicationResultBuildItem,
             BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer,
             List<ExceptionMapperBuildItem> mappers, Capabilities capabilities) {
         AdditionalBeanBuildItem.Builder beanBuilder = AdditionalBeanBuildItem.builder().setUnremovable();
         ExceptionMapping exceptions = ResteasyReactiveExceptionMappingScanner
@@ -118,6 +120,9 @@ public class ResteasyReactiveScanningProcessor {
         for (ExceptionMapperBuildItem additionalExceptionMapper : mappers) {
             if (additionalExceptionMapper.isRegisterAsBean()) {
                 beanBuilder.addBeanClass(additionalExceptionMapper.getClassName());
+            } else {
+                reflectiveClassBuildItemBuildProducer
+                        .produce(new ReflectiveClassBuildItem(true, false, false, additionalExceptionMapper.getClassName()));
             }
             int priority = Priorities.USER;
             if (additionalExceptionMapper.getPriority() != null) {
@@ -141,6 +146,7 @@ public class ResteasyReactiveScanningProcessor {
     @BuildStep
     public ParamConverterProvidersBuildItem scanForParamConverters(CombinedIndexBuildItem combinedIndexBuildItem,
             BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer,
             ApplicationResultBuildItem applicationResultBuildItem,
             List<ParamConverterBuildItem> paramConverterBuildItems) {
 
@@ -153,6 +159,9 @@ public class ResteasyReactiveScanningProcessor {
         for (ParamConverterBuildItem additionalParamConverter : paramConverterBuildItems) {
             if (additionalParamConverter.isRegisterAsBean()) {
                 beanBuilder.addBeanClass(additionalParamConverter.getClassName());
+            } else {
+                reflectiveClassBuildItemBuildProducer
+                        .produce(new ReflectiveClassBuildItem(true, false, false, additionalParamConverter.getClassName()));
             }
             int priority = Priorities.USER;
             if (additionalParamConverter.getPriority() != null) {
@@ -197,6 +206,7 @@ public class ResteasyReactiveScanningProcessor {
     public ContextResolversBuildItem scanForContextResolvers(CombinedIndexBuildItem combinedIndexBuildItem,
             ApplicationResultBuildItem applicationResultBuildItem,
             BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer,
             List<ContextResolverBuildItem> additionalResolvers) {
         IndexView index = combinedIndexBuildItem.getComputingIndex();
         AdditionalBeanBuildItem.Builder beanBuilder = AdditionalBeanBuildItem.builder().setUnremovable();
@@ -210,6 +220,9 @@ public class ResteasyReactiveScanningProcessor {
         for (ContextResolverBuildItem i : additionalResolvers) {
             if (i.isRegisterAsBean()) {
                 beanBuilder.addBeanClass(i.getClassName());
+            } else {
+                reflectiveClassBuildItemBuildProducer
+                        .produce(new ReflectiveClassBuildItem(true, false, false, i.getClassName()));
             }
             ResourceContextResolver resolver = new ResourceContextResolver();
             resolver.setClassName(i.getClassName());

